### PR TITLE
Bump Pywavelets to 1.0.0

### PR DESCRIPTION
**Pull request by sciunto**
## Description

We can discuss either we merge this PR now or after 0.17 is out. This PR corresponds to one of the tasks in the TODO list.
Pywavelets 1.0.0 was released in Aug 2018. https://github.com/PyWavelets/pywt/releases

It also applies the suggestion 2. in #3407

### CODE REVIEW

**Comment by sciunto** [comment]
The failure is unrelated.

**Comment by JDWarner** [comment]
Should we consider going all the way to 1.1 (basically pywt current release)?

1.1 is where they dropped Python 2, so thematically it seems like it would match our intent to do the same.  The changes from pywt 1.0 to 1.1 seem essentially uniformly positive.  The backwards incompatible changes are both actually bugfixes from what I can tell.

**Comment by emmanuelle** [comment]
@JDWarner what would we gain from bumping to 1.1 instead of 1.0? If the warnings disappear, I'm happy with the lowest version enabling this.

**Comment by rfezzani** [comment]
Minimum version set to 1.0 is fine for me.

**Comment by JDWarner** [comment]
Moving to 1.1 directly keeps us from indirectly creating a breaking change later when we do.

One of the changes in 1.1 is a fix where previously the `cwt` results replicated the results in Matlab's 2012a version.  Except Matlab was wrong.  The correct results per the literature reference paper are the complex conjugate of what Matlab 2012a and prior (as well as pywt < 1.1) returned.  Matlab has since fixed this.

The `cwt` wavelet in 1.1 also now has a FFT mode which dramatically improves performance.

**Comment by emmanuelle** [comment]
OK I see. Thanks for the update @JDWarner. If there are important bug fixes then it's different. Anyway it's not like asking people to upgrade their numpy or jupyter, it's a less frequent dependency so we won't mess up too much with people environments.

**Comment by sciunto** [comment]
> Anyway it's not like asking people to upgrade their numpy or jupyter, it's a less frequent dependency so we won't mess up too much with people environments.

@emmanuelle but pywt is in the default requirements. What makes the difference with numpy?

> Moving to 1.1 directly keeps us from indirectly creating a breaking change later when we do.

Does it mean that if I increase the min version, I have to do extra work on this PR? (I'm not sure to understand what you mean...)

**Comment by emmanuelle** [comment]
@sciunto the difference is that `pywt` is used by less projects than numpy (but checking on Github's "used by", pywt is still used by 16.4k projects... for comparison scikit-image is 35k so it's not a small number). Therefore the probability that people need to use an old version of pywt for a different project is quite small.

I think that @JDWarner is just suggesting to change to 1.1.1. At the moment we're not using `pywt.cwt` (from a quick grep, correct me if I'm wrong) so we're not breaking anything, but if we'd start using cwt then it would be a breaking change to switch to 1.0 to 1.1. Is this what you meant @JDWarner ?

**Comment by sciunto** [comment]
Thank you for the clarifications @emmanuelle I let @JDWarner confirms.

@scikit-image/core Do we have a consensus/majority to set the min version to 1.1.1 for pywavelets?

**Comment by grlee77** [comment]
I am +1 on bumping to 1.1. Functionality should be the same as what we have now, but it does allow some minor refactoring/cleanup as partially done in this PR. I would be happy to make a PR with the other suggested change from #3407 afterwards.

**Comment by rfezzani** [comment]
I am +1 for 1.1

File: skimage/restoration/tests/test_denoise.py [hunk 1]
```diff
@@ -898,12 +868,10 @@ def test_cycle_spinning_num_workers():
     func_kw = dict(sigma=sigma, multichannel=True, rescale_sigma=True)
 
     # same results are expected whether using 1 worker or multiple workers
-    with expected_warnings([PYWAVELET_ND_INDEXING_WARNING]):
-        dn_cc1 = restoration.cycle_spin(noisy, denoise_func, max_shifts=1,
+    dn_cc1 = restoration.cycle_spin(noisy, denoise_func, max_shifts=1,
                                         func_kw=func_kw, multichannel=False,
```
Comment (stefanv) [inline skimage/restoration/tests/test_denoise.py:872]: Just check indentation here.

**Comment by rfezzani** [comment]
:tada: Thank you @sciunto!

**Pull request merged**